patterns/contracts-as-note-owners/trustless-manager/src/main.nr

```diff
@@ -0,0 +1,164 @@
+    In this case, this file will contain the code for the TrustlessManager.
+    
+    Note: This is a simplified example. In a real world scenario, this would have more checks, conditions to be met, a way for Alice to cancel the escrow if Bob didn't meet the conditions, etc.
+*/
+
+    // Declaring the slots for the notes
+    global RECEIPT_NOTE_SLOT = 1;
+
+    // Declaring storage
+    struct Storage {
+        receipt_notes: Set<ReceiptNote, RECEIPT_NOTE_LEN>,
+    }
+
+    // Declaring storage implementation
+    impl Storage {
+        fn init(context: Context) -> Self {
+            Storage {
+                receipt_notes: Set::new(context, RECEIPT_NOTE_SLOT, ReceiptNoteMethods),
+            }
+        }
+    }
+ 
+    /// @notice Constructor for the contract.
+    #[aztec(private)]
+    fn constructor() {}
+
+    /// @notice Deposits the fee agreed upon between the parties in the escrow contract and emits a receipt note to the involved parties.
+    /// @param bob The address of the user who will receive the escrowed fee if the conditions are met.
+    /// @param fee The amount of tokens to pay.
+    /// @param escrow The address of the contract used to escrow the fee.
+    /// @param payment_token The address of the token Alice will pay the fee with.
+    /// @param nonce Alice's authwit nonce.
+    #[aztec(private)]
+    fn deposit_fee(bob: AztecAddress, fee: Field, escrow: AztecAddress, payment_token: AztecAddress, nonce: Field) {
+        // Store the sender as Alice for clarity.
+        let alice = context.msg_sender();
+
+        // Store the address of the contract for readability.
+        let address_this = context.this_address();
+
+        // Create a shared nullifier for the note
+        let shared_nullifier = rand();
+
+        // Create the array containing the recipients that will get the escrow note encrypted to them
+        let recipients: [AztecAddress; 4] = [alice, bob, AztecAddress::from_field(0), AztecAddress::from_field(0)];
+
+        // Instantiate the contract that will escrow the fee
+        let token_escrow = Escrow::at(escrow);
+
+        // Take the payment from Alice.
+        let random_id = token_escrow.escrow(
+            &mut context,
+            alice.to_field(),
+            address_this.to_field(),
+            fee,
+            payment_token.to_field(),
+            nonce,
+            recipients
+        );
+
+        // Create the note. We need the context to be able to push to the nullifier tree.
+        let mut receipt_note = ReceiptNote::new(address_this, alice, bob, random_id, shared_nullifier, payment_token, escrow, fee);
+
+        // Insert it into storage and broadcast it to alice.
+        storage.receipt_notes.insert(&mut receipt_note, true);
+
+        // Broadcast it to bob as well, so he can also call settle when the conditions are met
```

Comment:
shouldn't we do this in the broadcast?

Reply:
You mean broadcasting it to both in the `broadcast` fn? I think this is a bit more clear for anyone reading, but it could be done (we probably could do it in the private oracle) to avoid the ugly logic in the function body.

Reply:
yes, as we never insert without broacasting to both?